lib/compress/zstd_ldm.c

```diff
@@ -447,6 +467,17 @@ static size_t ZSTD_ldm_generateSequences_internal(
             ZSTD_ldm_insertEntry(ldmState, hash, newEntry, *params);
 
             anchor = split + forwardMatchLength;
+
+            /* If there is an overlapping match skip to the end of it.
+             * No need to insert every position into our table. This
+             * is especially slow because position of a long overlapping match
+             * is inserted, every position is inserted.
+             */
+            if (mLength > (size_t)offset) {
```

Comment:
If I understand correctly, we want to save work in the `while (ip < ilimit)` loop.

If that is the case, I would instead detect the degenerate case using `anchor > ip + hashed`, then reset `ip = anchor - hashed` and break instead of continue. It makes it a bit clearer where the savings are expected and also makes it easy to see the relation with the `ip` reset and the increment line 483.

Reply:
Thats a good point! I actually intended to continue the `while (ip < ilimit)` loop, not the inner loop, so this is a logic bug. What you're saying is what I was intending to write. Thanks for the review!

Reply:
That isn't quite the same logic, since `mLength > offset` detects overlapping matches, whereas `anchor > ip + hashed` detects long matches.

However, I've found that `anchor > ip + hashed` improves compression ratio by a reasonable amount in some `--patch-from` use cases that I tested, where the compression ratio is very high. So it is probably a reasonable choice.